bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go

```diff
@@ -919,6 +922,10 @@ func (a *GeneralController) computeStatusForExternalMetric(
 }
 
 func (a *GeneralController) recordInitialRecommendation(currentReplicas int32, key string) {
+	// add lock
+	a.recommendationsLock.Lock()
+	defer a.recommendationsLock.Unlock()
+
```

Comment:
4 functions (reconcileKey, storeScaleEvent, stabilizeRecommendationWB, convertDesiredReplicasWithBR) should add locks too.

Reply:
Thanks for your help, scaleUp/scaleDown events lock has been added.